pkg/curl/schemes.go

```diff
@@ -43,6 +44,15 @@ type FileScheme interface {
 	Fetch(u *url.URL) (io.ReaderAt, error)
 }
 
+// FileSchemeRetryFilter contains extra RetryFilter method for a FileScheme
+// wrapped by SchemeWithRetries.
+type FileSchemeRetryFilter interface {
+	// RetryFilter lets a FileScheme filter for errors returned by Fetch
+	// which are worth retrying. If this interface is not implemented, the
+	// default for SchemeWithRetries is to always retry.
+	RetryFilter(u *url.URL, err error) bool
```

Comment:
why does this take the url?

Reply:
Look at the implementation `func (s Schemes) RetryFilter(u *url.URL, err error)`. `Schemes` looks at the url to find the appropriate FileScheme and calls `RetryFilter` on it with the same arguments.

Reply:
can you annotate in the comment what true and false mean?

RetryFilters return true to mean the request should be retried.

Reply:
Done.